xbmc/filesystem/File.cpp

```diff
@@ -291,15 +285,15 @@ bool CFile::Open(const CURL& file, const unsigned int flags)
       if (m_flags & READ_CACHED)
       {
         // for internet stream, if it contains multiple stream, file cache need handle it specially.
-        m_pFile = new CFileCache(m_flags);
+        m_pFile = std::make_unique<CFileCache>(m_flags);
 
         if (!m_pFile)
           return false;
 
         return m_pFile->Open(url);
       }
     }
-    m_pFile = CFileFactory::CreateLoader(url);
+    m_pFile.reset(CFileFactory::CreateLoader(url));
```

Comment:
It might be more clear to user operator= instead of reset, e.g.
```c++
m_pFile = std::move(CFileFactory::CreateLoader(url));
```
(not sure if the std::move is required but I think it is)

Reply:
I was actually unaware of this, thanks for pointing that out!

Reply:
Yes, std::move is clearer here. Same for "m_pDemuxer.reset(CDVDFactoryDemuxer::CreateDemuxer(m_pInputStream));" above.

Reply:
it's not possible to use the `operator=()` in both instances as `.reset()` initialises the `unique_ptr` (that's why I was unaware, it's not possible :upside_down_face:).

```
/home/lukas/Documents/git/xbmc/xbmc/filesystem/File.cpp:1205:53: error: no match for ‘operator=’ (operand types are ‘std::unique_ptr<XFILE::IFile>’ and ‘std::remove_reference<XFILE::IFile*>::type’ {aka ‘XFILE::IFile*’})
 1205 |   m_file = std::move(CFileFactory::CreateLoader(url));
      |                                                     ^
```

Both `CFileFactory::CreateLoader` and `CDVDFactoryDemuxer::CreateDemuxer` return raw pointers which are used to initialise the unique_ptr with `.reset()`

If they returned a `unique_ptr` this would be possible but I think that's outside the scope of this PR and can be included in a future PR.

Reply:
ok got it. future PR sounds good if you get to it